src/game/SpellEffects.cpp

```diff
@@ -4817,8 +4809,8 @@ void Spell::EffectSummonDeadPet(SpellEffectIndex /*eff_idx*/)
         return;
     if (pet->isAlive())
         return;
-    if (damage < 0)
-        return;
+
+    ((Unit*)pet)->NearTeleportTo(m_caster->GetPositionX(), m_caster->GetPositionY(), m_caster->GetPositionZ(), m_caster->GetOrientation());
```

Comment:
Would this perhaps be even better?
>float px, py, pz;
m_caster->GetClosePoint(px, py, pz, pet->GetObjectBoundingRadius());
((Unit*)pet)->NearTeleportTo(px, py, pz, m_caster->GetOrientation());

I'm thinking of changing all summons to follow that so they don't summon right at the location of the caster which looks pretty stupid.